BnbnavNetClient/Views/MapView.axaml.cs

```diff
@@ -92,18 +93,21 @@ protected override void OnInitialized()
 
             var seenEdges = new List<Edge>();
             MapViewModel.ContextMenuItems.Clear();
-            MapViewModel.ContextMenuItems.AddRange(HitTest(pointerPos).Select(x =>
+            MapViewModel.ContextMenuItems.AddRange(HitTest(pointerPos).SelectMany(x =>
             {
                 switch (x)
                 {
                     case Node node:
-                        return new MenuItem
+                        return new MenuItem[]
                         {
-                            Header = _i18n["NODE_DELETE"],
-                            Command = ReactiveCommand.Create(() =>
+                            new()
                             {
-                                MapViewModel.QueueDelete(node);
-                            })
+                                Header = _i18n["NODE_DELETE"],
+                                Command = ReactiveCommand.Create(() =>
+                                {
+                                    MapViewModel.QueueDelete(node);
+                                })
+                            }
```

Comment:
consider using some converter/data template here instead, though this works

Reply:
This can be refactored later by someone who knows what they are doing